src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -294,21 +294,24 @@ public AzureVMDeploymentInfo createDeployment(
             putVariable(tmp, "cloudTag", template.getAzureCloud().getCloudName());
 
             // add purchase plan for image if needed in reference configuration
-            if (!useCustomImage && StringUtils.isBlank(template.getImageId())) {
-                VirtualMachineImage image = azureClient.virtualMachineImages().getImage(locationName,
-                        template.getImagePublisher(), template.getImageOffer(), template.getImageSku(), template.getImageVersion());
-                if (image != null) {
-                    PurchasePlan plan = image.plan();
-                    if (plan != null) {
-                        ArrayNode resources = (ArrayNode) tmp.get("resources");
-                        for (JsonNode resource : resources) {
-                            String type = resource.get("type").asText();
-                            if (type.contains("virtualMachine")) {
-                                ObjectNode planNode = mapper.createObjectNode();
-                                planNode.put("name", plan.name());
-                                planNode.put("publisher", plan.publisher());
-                                planNode.put("product", plan.product());
-                                ObjectNode.class.cast(resource).replace("plan", planNode);
+            if (!isBasic && !useCustomImage && StringUtils.isBlank(template.getImageId())) {
+                boolean isImageParameterValid = checkImageParameter(template);
+                if (isImageParameterValid) {
```

Comment:
what is image parameters not valid? continue flow or error out?

Reply:
if parameters are not valid, we can continue the flow since what we want to do here is to add more info to the ARM template. The provision won't succeed whether we add the plan info.